ga4gh/backend.py

```diff
@@ -18,14 +20,34 @@ class Backend(object):
     """
     def __init__(self, dataDir, variantSetClass):
         self._dataDir = dataDir
+        # Variants
         self._variantSetIdMap = {}
-        # All directories in datadir are assumed to correspond to VariantSets.
-        for variantSetId in os.listdir(self._dataDir):
-            relativePath = os.path.join(self._dataDir, variantSetId)
+        variantSetDir = os.path.join(self._dataDir, "variants")
```

Comment:
It seems like you're repeating yourself a lot here over the three object types... this could be refactored.